src/main/java/ru/ayaz/UserHandler.java

```diff
@@ -2,130 +2,42 @@
 
 import java.io.BufferedReader;
 import java.io.IOException;
-import java.io.InputStreamReader;
-import java.io.PrintWriter;
-import java.net.Socket;
 
 
 public class UserHandler implements Runnable {
```

Comment:
&#x1F34E; **Issue**
I believe you should initialize all private fields of this class in the constructor.
Better yet, you can pass in a socket and figure out how to extract streams from it inside UserHandler.